Add "pkgs.k8s.io: Introducing Kubernetes community-owned package repositories" blog post (#42023)
kubernetes/website: content/en/blog/_posts/2023-08-15-pkgs-k8s-io-introduction.md

---
layout: blog

> title: "pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"

Reviewer: ```suggestion
title: Kubernetes 1.28: "pkgs.k8s.io Introducing Kubernetes Community-owned Package Repositories"
```
Suggestions:
- Title case for title
- The previous blog posts have the version starting the title

Reviewer 2: Hey @reylejano, this is an out of tree change so we intentionally removed the prefix. Is that fine for you, too?

Reviewer 3: Release comms / @bradmccoydev - how do you want to handle this announcement? Do you want it to be part of the v1.28 release comms or separate from it?

I'd assumed that there _is_ a link. You can still set a custom title; if you do, I strongly suggest putting this blog article as the very first or very last post-release blog article for v1.28

Reviewer 4: Hi sure no problem at all, I have updated the Comms Board for it to go out the same day

Reviewer 3: Thanks @bradmccoydev

> date: 2023-08-15

Reviewer 3: Release comms need to confirm the date here.

Author: @ramrodo was following up with me on this and can confirm if that's the target date. I can update it once we have written confirmation

Reviewer: Currently, there are 2 1.28 feature blogs proposed to be published for August 15
- Kubernetes 1.28: Non-Graceful Node Shutdown Moves to GA
- pkgs.k8s.io: Introducing Kubernetes community-owned package repositories

Can the 1.28 comms team confirm the publication date 
@bradmccoydev @krol3 @mashby2022 @ramrodo @thschue

Reviewer 4: Can we please put the date to 2023-08-22 please @xmudrii. Thanks

Reviewer 2: Can we release earlier than Aug 15? The blog post is a dependency for the docs, mean we can't publish docs without the blog post.

Reviewer 3: We can _merge_ the article earlier. I'm aiming to take care of both details (blog article and associated docs changes).

Reviewer 3: Confirmed as 2023-08-15. Let's set the _time_ to be later than the official release announcement.

Reviewer 2: Good, thanks!

slug: pkgs-k8s-io-introduction
---

**Author**: Marko Mudrinić (Kubermatic)

> We're very excited to introduce the Kubernetes community-owned package

Reviewer 3: ```suggestion
On behalf of Kubernetes SIG Release, I am very excited to introduce the
Kubernetes community-owned software
```

repositories for Debian and RPM packages: `pkgs.k8s.io`! The new package
repositories are replacement for the Google-hosted package repositories
(`apt.kubernetes.io` and `yum.kubernetes.io`) that we've been using since
Kubernetes v1.5.

This blog post contains information about these new package repositories,
what does it mean to you as an end user, and how to migrate to the new
repositories.

## What you need to know about the new package repositories?

> - This is an **opt-in change**, you're required to manually migrate from the

Reviewer 3: ```suggestion
- This is an **opt-in change**; you're required to manually migrate from the
```

  Google-hosted repository to the Kubernetes community-owned repositories.

>   See below for migration information and instructions.

Reviewer 3: ```suggestion
  See [how to migrate](#how-to-migrate) later in this announcement for migration information
  and instructions.
```

- Access to the Google-hosted repository will remain intact for the foreseeable
  future. However, the Kubernetes project plans to stop publishing packages to
  the Google-hosted repository in the future. The project strongly recommends
  migrating to the Kubernetes package repositories going forward.
- The Kubernetes package repositories contain packages beginning with those
  Kubernetes versions that were still under support when the community took
  over the package builds. This means that anything before v1.24.0 will only be
  available in the Google-hosted repository.
- There's a dedicated package repository for each Kubernetes minor version.

>   When upgrading to to a different minor release, you must bear in mind that

Reviewer 5: ```suggestion
  When upgrading to a different minor release, you must bear in mind that
```

with technical support to integrate OBS with our release processes.

We use SUSE's OBS instance for building and publishing packages. Upon building
a new release, our tooling automatically pushes needed artifacts and 
package specifications to `build.opensuse.org`. That will trigger the build
process that's going to build packages for all supported architectures (AMD64,
ARM64, PPC64LE, S390X). At the end, generated packages will be automatically
pushed to our community-owned S3 bucket making them available to all users.

We want to take this opportunity to thank SUSE for allowing us to use
`build.opensuse.org` and their generous support to make this integration
possible!

## What are significant differences between the Google-hosted and Kubernetes package repositories?

There are three significant differences that you should be aware of:

- There's a dedicated package repository for each Kubernetes minor release.
  For example, repository called `core:/stable:/v1.28` only hosts packages for
  stable Kubernetes v1.28 releases. This means you can install v1.28.0 from
  this repository, but you can't install v1.27.0 or any other minor release

>   other then v1.28. Upon upgrading to another minor version, you have to add a

Reviewer 5: ```suggestion
  other than v1.28. Upon upgrading to another minor version, you have to add a
```

  - Kubernetes repositories for v1.28 and onwards are going to have published
    only versions that are used by that Kubernetes minor release
    - Speaking of v1.28, only kubernetes-cni 1.2.0 and cri-tols v1.28 are going
      to be available in the repository for Kubernetes v1.28
    - Similar for v1.29, we only plan on publishing cri-tools v1.29 and
      whatever kubernetes-cni version is going to be used by Kubernetes v1.29
- The revision part of the package version (the `-00` part in `1.28.0-00`) is
  now autogenerated by the OpenBuildService platform and has a different format.
  The revision is now in the format of `-x.y`, e.g. `1.28.0-1.1`

## Does this in any way affect existing Google-hosted repositories?

The Google-hosted repository and all packages published to it will continue
working in the same way as before. There are no changes in how we build and
publish packages to the Google-hosted repository, all newly-introduced changes
are only affecting packages publish to the community-owned repositories.

However, as mentioned at the beginning of this blog post, we plan to stop
publishing packages to the Google-hosted repository in the future.

> ## How to migrate to the Kubernetes community-owned repositories?

Reviewer 3: ```suggestion
## How to migrate to the Kubernetes community-owned repositories? {#how-to-migrate}
```

> ### Debian, Ubuntu, and operating systems using `apt`/`apt-get`

Reviewer 3: ```suggestion
### Debian, Ubuntu, and operating systems using `apt`/`apt-get` {#how-to-migrate-deb}
```

1. Replace the `apt` repository definition so that `apt` points to the new
   repository instead of the Google-hosted repository. Make sure to replace the

>    Kubernetes minor version in the command below with your the minor version

Reviewer 5: ```suggestion
   Kubernetes minor version in the command below with the minor version
```

   that you're currently using:

   ```shell
   echo "deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.28/deb/ /" | sudo tee /etc/apt/sources.list.d/kubernetes.list
   ```

2. Download the public signing key for the Kubernetes package repositories.

>    The same signing key is used for all repositories so you can disregard the

Reviewer 5: ```suggestion
   The same signing key is used for all repositories, so you can disregard the
```

   version in the URL:

   ```shell
   curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
   ```

3. Update the `apt` package index:

  ```shell
  sudo apt-get update
  ```

> ### CentOS, Fedora, RHEL, and operating systems using `rpm`/`dnf`

Reviewer 3: ```suggestion
### CentOS, Fedora, RHEL, and operating systems using `rpm`/`dnf` {#how-to-migrate-rpm}
```

1. Replace the `yum` repository definition so that `yum` points to the new 
   repository instead of the Google-hosted repository. Make sure to replace the

>    Kubernetes minor version in the command below with your the minor version

Reviewer 5: ```suggestion
   Kubernetes minor version in the command below with the minor version
```

---
layout: blog
title: "pkgs.k8s.io: Introducing Kubernetes Community-Owned Package Repositories"

> date: 2023-08-15T15:00:00+0000

Reviewer 3: Updating the date to match when the release actually happened
```suggestion
date: 2023-08-15T20:00:00+0000
```
